pkg/api/types.go

```diff
@@ -921,6 +921,9 @@ type PodSpec struct {
 	// used must be specified.
 	// Optional: Default to false.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// Run the container using the host ipc mode.
+	// Optional: Default to false.
+	HostIPC bool `json:"hostIPC,omitempty"`
```

Comment:
I can't tell if this thread already asked and answered this question, but this PR lets us choose between "one (generated) namespace per container" and "host namespace", but it would also be valuable to let containers share a (non-host) IPC namespace with another container in their pod. Will this pod-level boolean make sense if we eventually have a per-container IPC namespace option?